cloud/azure/main.tf

```diff
@@ -239,14 +201,11 @@ resource "azurerm_postgresql_database" "civiform" {
   collation           = "English_United States.1252"
 }
 
-resource "azurerm_postgresql_firewall_rule" "civiform" {
-  name                = "civiform-db-firewall"
-  resource_group_name = azurerm_resource_group.rg.name
-  server_name         = azurerm_postgresql_server.civiform.name
 
-  # TODO: replace with something more specific
-  # All zeros here configures the filewall to accept all incoming connections
-  # from anywhere inside Azure.
-  start_ip_address = "0.0.0.0"
-  end_ip_address   = "0.0.0.0"
-}
+resource "azurerm_postgresql_virtual_network_rule" "civiform" {
+  name                                 = "sqlvnetrule"
+  resource_group_name                  = azurerm_resource_group.rg.name
+  server_name                          = azurerm_postgresql_server.civiform.name
+  subnet_id                            = azurerm_subnet.server_subnet.id
+  ignore_missing_vnet_service_endpoint = true
```

Comment:
Question (blocking): does this PR still work if `ignore_missing_vnet_service_endpoint` is set to false? False seems preferable if so.

Reply:
updated this to False